lib/bademagic_module/utils/image_utils.dart

```diff
@@ -170,18 +150,31 @@ class ImageUtils {
   //function to generate the view for the Dialog from the given asset
   Future<ui.Image> generateImageView(String asset) async {
     await _loadSVG(asset);
-    final ui.Image scaledImage = await _scaleSVG(30, 120);
+    ui.Image image =
+        await picture.toImage(originalWidth.toInt(), originalHeight.toInt());
+    final ui.Image scaledImage = await _scaleSVG(image, 30, 120);
     return _trimSVG(scaledImage);
   }
 
   //function to generate the LED hex from the given asset
   Future<List<String>> generateLedHex(String asset) async {
```

Comment:
Why not start working on bytes rather than Strings?

Reply:
Adds an extra layer of encryption/decryption

Reply:
It is going to be very complicated in dart there is a lot of things that can get way too complicated inorder to do so and maybe it increase the computation time.

Reply:
maybe this is something that we can work on afterwards if you agree.

Reply:
Sure thing, We can always create a great framework around it to manage complexity. Let's park this as an improvement we can look at after we finish the app